SERVER/src/controller/user.js

```diff
@@ -4,33 +4,68 @@ import bcrypt from 'bcrypt';
 import userHelpers from '../helpers/user';
 import data from '../data';
 
-let counter = 0;
+import db from '../models/index';
+
+const { pool } = db;
+
+const counter = 0;
 let homeAddressCounter = 0;
 let jobCounter = 0;
 class User {
-  static createAccount(req, res) {
-    const status = 'unverified'; let token;
+  static async createAccount(req, res) {
+    let token;
     const {
       firstname, lastname, email, phone, password, isAdmin,
     } = req.body;
-    const user = userHelpers.findUser(data.users, email, 'email');
+    const user = await userHelpers.findUser(email);
     if (user.exist) {
       res.status(409).json({ error: 'user already exists' }).end();
     } else {
-      counter += 1;
-      token = jwt.sign({ data: counter }, 'landxxxofxxxopportunity', { expiresIn: '24h' });
-      data.users.push({
-        id: counter, firstname, lastname, email, phone, password, status, isAdmin,
-      });
-      res.status(201).json({
-        status: 201,
-        data: {
-          token, id: counter, firstname, lastname, email, phone, password, status, isAdmin,
-        },
-      }).end();
+      token = jwt.sign({ data: email }, 'landxxxofxxxopportunity', { expiresIn: '24h' });
+      const client = await pool.connect();
+      const query = {
+        text: 'INSERT INTO users (firstname, lastname, email, phone, password, isAdmin ) VALUES($1, $2, $3, $4, $5, $6) RETURNING *',
+        values: [firstname, lastname, email, phone, password, isAdmin],
+      };
+      try {
+        const userData = await client.query(query);
+        res.status(201).json({
+          status: 201,
+          data: {
+            token, ...userData.rows[0],
+          },
+        }).end();
+      } catch (err) {
+        res.status(500).json({ error: err.message }).end();
+      } finally {
+        await client.release();
+      }
     }
   }
 
+  static authenticate(req, res) {
+    const { email, password } = req.body;
+    const findUserData = userHelpers.findUser(data.users, email, 'email');
+    if (findUserData.exist) {
+      const hash = findUserData.data.password;
+      bcrypt.compare(password, hash, (err, result) => {
+        const token = jwt.sign({ data: findUserData.data.id }, 'landxxxofxxxopportunity', { expiresIn: '24h' });
```

Comment:
Do your token signing when passwords comparison is successful.
You should save your token secret in an environment variable.